### ISSUE TITLE
Input Image is set globally? Other connected clients can see them

### ISSUE THREAD

**Issue by 666spawn999** [lllyasviel/Fooocus#716]
Hi,
I'm trying to setup a Fooocus intance for me and my colleagues. But Input Images are shared and kept in cache somehow...

**Error**
- Client A adds an Input Image into "Upscale" or "Inpaint" Tab
- Client B open the "Inpaint" Tab -> The Image of Client A appears
- Image stays in Cache and is showed even after reloads, browser changes etc... 

Short Video of this error with 2 browsers side by side
![Fooocus_InputImage_Sharing](https://github.com/lllyasviel/Fooocus/assets/33587160/90dda8bd-96c7-4519-ac5d-56afa484e3ac)

**Steps to Reproduce**
- Open 2 Browser windows (same/different Browsers or incognito, doesn't matter)
- Client A adds an image to the "Upscale" InputImage
- Client B opens the "Inpaint" Tab --> Image of Client A is shown
- Open a completely different Browser -> Image will be loaded in "Inpaint" Tab

**Some thoughts**
- I think this may be a problem of the global/"shared" Variables in the code? The Input Image seems to be cached globally instead of being bound to a session. 

A fix or ony tips how remove this behaviour would be highly appreciated!
For people who are working alone with Fooocus, this may be not a problem. But for Privacy reasons in multi-user scenarios, this is a critical issue ... 

(I really love Fooocus and love all the features and the simplicity of it. Thanks for all your work! 
But sadly, bugs like this prevent me from using it in production or teams :( )

**Comment by tarik23** [comment]
+1

**Comment by 666spawn999** [comment]
@tarik23 That is the other problem, that I have reported:
https://github.com/lllyasviel/Fooocus/issues/713

**Comment by mashb1t** [comment]
The following code solves the image issue, but there might be a dynamic solution for calling the methods select_uov_tab etc. as fn can not handle static input strings and lambda can't handle non-global state inputs.

Maybe you find a better solution or we can merge it as is.

```         
            current_tab = gr.Textbox(value='uov', visible=False)

            default_image = gr.State(value=None)

            def update_default_image(x):
                if isinstance(x, dict):
                    default_image = x['image']
                else:
                    default_image = x
                return default_image

            def clear_default_image():
                return None

            uov_input_image.upload(update_default_image, inputs=uov_input_image, outputs=default_image, queue=False)
            inpaint_input_image.upload(update_default_image, inputs=inpaint_input_image, outputs=default_image, queue=False)

            uov_input_image.clear(clear_default_image, outputs=default_image, queue=False)
            inpaint_input_image.clear(clear_default_image, outputs=default_image, queue=False)

            def select_uov_tab(image):
                return 'uov', image

            def select_inpaint_tab(image):
                return 'inpaint', image

            def select_ip_tab(image):
                return 'ip', image

            uov_tab.select(fn=select_uov_tab, inputs=[default_image], outputs=[current_tab, uov_input_image], queue=False, _js=down_js)
            inpaint_tab.select(fn=select_inpaint_tab, inputs=[default_image], outputs=[current_tab, inpaint_input_image], queue=False, _js=down_js)
            ip_tab.select(fn=select_ip_tab, inputs=[default_image], outputs=[current_tab], queue=False, _js=down_js)

```

**Comment by lllyasviel** [comment]
@mashb1t thanks for this!
if you make a commit i will merge yours 
if not then i will copy this

**Comment by mashb1t** [comment]
will create a PR during the day 👍

**Comment by 666spawn999** [comment]
thanks for looking into this!

I tried it and it seems that this does the trick!

I myself looked into Gradio States etc. but I didn't got it right. Thanks for this and I think this can be closed when the PR is accepted. 

Thanks again! :D

**Comment by lllyasviel** [comment]
fixed in 2.1.758

**Comment by mashb1t** [comment]
@666spawn999 issue can be closed
